test/distributed/_tensor/debug/test_comm_mode_features.py

```diff
@@ -0,0 +1,359 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+# Owner(s): ["oncall: distributed"]
+
+from typing import Any, Dict
+
+import torch
+from torch.distributed._tensor import DeviceMesh
+from torch.distributed._tensor.api import DTensor
+from torch.distributed._tensor.debug import CommDebugMode
+from torch.distributed.tensor.parallel import (
+    ColwiseParallel,
+    parallelize_module,
+    RowwiseParallel,
+)
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    MLPModule,
+    MLPStacked,
+    ModelArgs,
+    NUM_DEVICES,
+    Transformer,
+    with_comms,
+)
+
+
+c10d_functional = torch.ops.c10d_functional
+
+
+class PrintModuleParamaterShardingTest(DTensorTestBase):
+    # checks if parameter / sharding info is the same as ground truth
+    def same_set_of_keys(self, dict1, dict2):
```

Comment:
call it something like `check_same_set_of_keys` because we are doing assertions inside

Reply:
do we plan to remove some duplicate code in the examples folder?

Reply:
I want the test cases to mirror the example folder so that there is a tutorial for users to learn how to use comm_mode_debug and I can use it to experiment as I add new features. The test file will be there to verify future changes doesn't break correct code